engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -162,8 +163,19 @@ public void deleteProcess(final ProcessRecord processRecord) {
 
     if (latestVersion == processRecord.getVersion()) {
       digestByIdColumnFamily.deleteIfExists(fkProcessId);
+
+      final var previousVersion = new AtomicLong(0);
+      processByIdAndVersionColumnFamily.whileEqualPrefix(
+          processId,
+          (key, value) -> {
+            final long version = key.second().getValue();
+            if (version > previousVersion.get()) {
+              previousVersion.set(version);
+            }
+          });
```

Comment:
❓ Will this always find a version, or could it result in `0`?

If it can result in `0`, we must handle this case in the version manager.